src/flexspi/peripheral.rs

```diff
@@ -0,0 +1,609 @@
+                    return Err(WaitCommandError::ArbiterIdle.into());
+                }
+
+                // Wait for an event before checking again to save power.
+                cortex_m::asm::wfe();
+            }
+        }
+    }
+
+    /// Clear the entire RX fifo.
+    pub fn clear_rx_fifo(&mut self) {
+        let flex_spi = unsafe { pac::Flexspi::steal() };
+        flex_spi.iprxfcr().modify(|_, w| w.clriprxf().set_bit());
+    }
+
+    /// Clear the entire TX fifo.
+    pub fn clear_tx_fifo(&mut self) {
+        let flex_spi = unsafe { pac::Flexspi::steal() };
+        flex_spi.iptxfcr().modify(|_, w| w.clriptxf().set_bit());
+    }
+
+    /// Write data to the IP TX FIFO and mark it ready for sending.
+    ///
+    /// This copies data into the IP TX FIFO, up to the watermark level.
+    pub fn fill_tx_fifo(&mut self, buffer: &[u8]) -> usize {
+        // Limit buffer to fifo length.
+        let watermark = self.get_tx_fifo_watermark_bytes();
+        let copy_len = buffer.len().min(watermark);
+        let buffer = &buffer[..copy_len];
+
+        unsafe {
+            let flex_spi = pac::Flexspi::steal();
+
+            // Copy the data from the FIFO to the user buffer.
+            read_u8_write_u32_volatile(buffer, flex_spi.tfdr(0).as_ptr().cast());
+
+            // Clear the TX watermark empty interrupt to transmit the data.
+            flex_spi.intr().write(|w| w.iptxwe().clear_bit_by_one());
+        }
+
+        // Report the number of bytes copied into the buffer.
+        copy_len
+    }
+
+    /// Drain the RX fifo.
+    ///
+    /// Returns the number of bytes written into the buffer.
+    ///
+    /// You should ensure that either the FIFO is filled to the watermark level,
+    /// or this is the final drain in a transfer.
+    ///
+    /// This function will always remove a full watermark of data from the FIFO,
+    /// since that is the only way to remove any data from the FIFO.
+    pub fn drain_rx_fifo(&mut self, buffer: &mut [u8]) -> usize {
+        let flex_spi = unsafe { pac::Flexspi::steal() };
+
+        // Get the fill level, but limit it to the watermark level.
+        let watermark = self.get_rx_fifo_watermark_bytes();
+        let fill_level = self.get_rx_fill_level_bytes();
+
+        // Truncate buffer to available FIFO size.
+        let copy_len = buffer.len().min(fill_level).min(watermark);
+        let buffer = &mut buffer[..copy_len];
+
+        // Copy data from TX FIFO to buffer.
+        unsafe {
+            read_u32_volatile_write_u8(flex_spi.rfdr(0).as_ptr().cast(), buffer);
+        }
+
+        // Pop watermark data from the FIFO.
+        flex_spi.intr().write(|w| w.iprxwa().clear_bit_by_one());
+
+        // Return the number of bytes written to the buffer.
+        copy_len
+    }
+
+    /// Discard the AHB RX buffer.
+    ///
+    /// Warning! Currently not implemented.
+    pub fn clear_ahb_rx_buffers(&mut self) {
+        // TODO: Uhhh... how?
```

Comment:
If anyone knows how to do this I would be very happy. The FlexSPI manual just mentions you have to clear the AHB RX buffer after an erase or write, but not how :]